### ISSUE TITLE
[Feature] Recursive context resolution

### ISSUE THREAD

**Issue by stefanomondino** [synesthesia-it/Murray#39]
In some contexts, it could come in handy to have recursive resolution of strings against current context, where context is a mix of local values (example: the `name` parameter from shell command) and environment values in `Murrayfile.json`.

Environment values may contain other placeholders and used as "variables" when lots of templates are created.

Use case:
MurrayFile:
```json
"environment": {
  "myCustomPath": "innerFolder/{{name|firstUppercase}}/{{name}}"
}
```

and BoneFile:
```json
{
"to": "{{myCustomPath}}.swift"
}
```

### Possible Approach:

A string should be recursively re-resolved only when Stencil instruction are contained within. However, the quickest approach is to call `resolve` again on a just-resolved string if its contents have changed after resolution.
In other words, we could have a pseudo method like this:
```swift
extension String {
 func resolve(against context: Context) {
 let resolved = self.properResolution(context: context) //the already existing method
    if resolved == self {
       return self
     } else {
        return resolved.resolve(against: context)
     }
}
}
```
